src/cursor/abstract_cursor.ts

```diff
@@ -837,16 +821,17 @@ function cleanupCursor(
       }
     }
 
-    return callback();
+    return;
   }
 
-  function completeCleanup() {
+  async function completeCleanup() {
```

Comment:
I had envisioned that when we finished the async-await cursor migration, we'd get rid of `cleanupCursor`.  Is there a reason we didn't tackle it now?

Reply:
It does not appear to be a straightforward change (8 usages with different arguments) but it is also something that can be done separately from making everything async.